go-controller/pkg/ovn/controller/apbroute/repair.go

```diff
@@ -361,3 +377,67 @@ func (c *ExternalGatewayMasterController) buildOVNECMPCache() map[string][]*ovnR
 	}
 	return ovnRouteCache
 }
+
+// returns a hybrid cache of map[podIPs]nodeName
+func (c *ExternalGatewayMasterController) buildOVNHybridCache() map[string]string {
+	ovnHybridCache := make(map[string]string)
+	p := func(item *nbdb.LogicalRouterPolicy) bool {
+		return item.Priority == types.HybridOverlayReroutePriority
+	}
+
+	logicalRouterPolicies, err := c.nbClient.findLogicalRouterPoliciesWithPredicate(p)
+	if err != nil {
+		klog.Errorf("CleanHybridPRoutes: failed to list hybrid routes: %v", err)
+		return nil
+	}
+
+	var foundNextHops []string
+	for _, lrp := range logicalRouterPolicies {
+		foundNextHops = append(foundNextHops, lrp.Nexthops...)
+	}
+
+	r := func(item *nbdb.LogicalRouterPort) bool {
+		for _, nexthop := range foundNextHops {
+			for _, ip := range item.Networks {
+				// grab only IP prefix and not mask
+				p := strings.Split(ip, "/")
+				if p[0] == nexthop {
+					return true
+				}
+			}
+		}
+		return false
+	}
+
+	grPorts, err := c.nbClient.findLogicalRouterPortWithPredicate(r)
+	if err != nil {
+		klog.Errorf("CleanHybridPRoutes: failed to search for logical router port: %v", err)
+		return nil
+	}
+
+	for _, grPort := range grPorts {
+		nodeName := strings.TrimPrefix(grPort.Name, types.GWRouterToJoinSwitchPrefix+types.GWRouterPrefix)
+		if len(nodeName) == 0 {
```

Comment:
should we check for `nodeName != grPort.Name` after trim prefix to be on the safe side?

Reply:
sure